crates/subspace-farmer/src/bin/subspace-farmer/commands/farm.rs

```diff
@@ -144,19 +144,25 @@ where
 
     let (run_cache_sync_tx, run_cache_sync_rx) = mpsc::channel::<SegmentIndex>(1);
 
-    tokio::spawn({
-        let piece_cache = piece_cache.clone();
-        let piece_getter = piece_getter.clone();
+    tokio::spawn(run_future_in_dedicated_thread(
```

Comment:
You don't need to spawn it, instead, if you don't want to wait for the finish, just store it in local variable. That way when this future (returned by `farm_multi_disk`) will also cancel corresponding task and stop dedicated thread automatically.

Look at `run_future_in_dedicated_thread` implementation.